contracts/src/TeleporterTokenDestination.sol

```diff
@@ -93,18 +93,33 @@ abstract contract TeleporterTokenDestination is
         );
         require(input.recipient != address(0), "TeleporterTokenDestination: zero recipient address");
 
-        // If the destination blockchain is the source bridge instance's blockchain,
-        // the destination bridge address must match the token source address.
+        // If the destination blockchain is the source blockchain,
+        // no multihop is needed. Only the required gas limit for the Teleporter message back to
+        // `sourceBlockchainID` is needed, which is provided by `input.requiredGasLimit`.
+        // Else, there will be a multihop transfer to the final destination.
+        // The first hop back to `sourceBlockchainID` requires `MULTIHOP_REQUIRED_GAS`,
+        // and the second hop to the final destination requires `input.requiredGasLimit`.
+        uint256 firstHopRequiredGas = input.requiredGasLimit;
+        uint256 secondHopRequiredGas;
```

Comment:
nit: can we initialize this to 0, so that it's clear that this variable contains a valid value for the non-multihop case? With it uninitialized, it kind of reads like this value will be set in both of the below if-else branches.

Reply:
I know at some point we had made the decision not to initialize variables like this to 0. I think it may have been a gas consideration? But I would be very surprised if the compiler didn't handle that... I do definitely favour explicit initialization.

Reply:
yup this is from previous audit suggestions as a gas saving

Reply:
Awesome, definitely not worth increasing the gas cost.